plugins/main/bridge/bridge.go

```diff
@@ -242,8 +253,21 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return err
 	}
 
+	// Determine the plugin type
+	// TODO: This assumes that for dual-stack, the plugin type is the
+	// same for IPv4 vs. IPv6. To allow for different plugin types,
+	// the IPv4 and IPv6 IPAM config will need to be split up and
+	// passed to each plugin separately, with results being combined.
+	var plugin_type string
+	switch {
+	case n.IPAM.Type != "":
+		plugin_type = n.IPAM.Type
+	default:
+		plugin_type = n.IPAM6.Type
+	}
```

Comment:
I was trying to minimize the complexity of the current review, and add the handling for this special case (dual-stack, but with different ipam plugins for v4 vs v6) in a followup pull request.